src/gluonts/model/deepstate/_estimator.py

```diff
@@ -174,6 +187,42 @@ def __init__(
         assert embedding_dimension is None or all(
             e > 0 for e in embedding_dimension
         ), "Elements of `embedding_dimension` should be > 0"
+        assert (
+            noise_std.lower_bound > 0
+        ), "Lower bound of noise standard deviation should be positive."
+        assert (
+            prior_cov.lower_bound > 0
+        ), "Lower bound of (diagonal of) prior covariance matrix should be positive."
+        assert (
+            innovation.lower_bound > 0
+        ), "Lower bound of innovation strength should be positive."
+
+        # For simplicity we do not allow infinity as upper bound here.
+        assert all(
+            np.isfinite(
+                [
+                    noise_std.lower_bound,
+                    prior_cov.lower_bound,
+                    innovation.lower_bound,
+                    noise_std.upper_bound,
+                    prior_cov.upper_bound,
+                    innovation.upper_bound,
+                ]
+            )
```

Comment:
All this could be made more compact with a list comprehension similar to a few lines below